hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -880,137 +797,143 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
+            maker_price = (floor(Decimal(maker_price) / price_quantum) ) * price_quantum
+
+            return maker_price
         else:
+
             price_quantum = maker_market.c_get_order_price_quantum(
                 market_pair.maker.trading_pair,
                 top_ask_price
             )
-            next_price = (round(Decimal(top_ask_price) / price_quantum) - 1) * price_quantum
+            next_price_below_top_ask = (round(Decimal(top_ask_price) / price_quantum) - 1) * price_quantum
 
-            # Calculate the order size limit from maker and taker market balances.
-            maker_balance_size_limit = maker_market.c_get_available_balance(market_pair.maker.base_asset)
-            taker_balance_size_limit = (taker_market.c_get_available_balance(market_pair.taker.quote_asset) /
-                                        float(next_price) *
-                                        self._order_size_taker_balance_factor)
+            try:
+                taker_price = taker_order_book.c_get_vwap_for_volume(True,
+                                                                     size).result_price
+            except ZeroDivisionError:
+                return None
 
-            # Convert the proposed maker order price to the equivalent price on the taker market.
-            adjusted_taker_price = (self._exchange_rate_conversion.adjust_token_rate(
-                market_pair.maker.quote_asset,
-                float(next_price)
-            ) / self._exchange_rate_conversion.adjust_token_rate(
-                market_pair.taker.quote_asset,
-                1.0
-            ))
-
-            # Calculate the order size limit from the minimal profitable hedge on the taker market.
-            profitable_hedge_price = adjusted_taker_price / (1 + self._min_profitability)
-            taker_order_book_size_limit = (
-                taker_order_book.c_get_volume_for_price(True, profitable_hedge_price).result_volume *
-                self._order_size_taker_volume_factor
-            )
+            if market_pair.maker.quote_asset != market_pair.taker.quote_asset:
+                taker_price *= self._exchange_rate_conversion.convert_token_value(1,
+                                                                                  market_pair.taker.quote_asset,
+                                                                                  market_pair.maker.quote_asset)
+
+            # You are buying on the taker market and selling on the maker market
+            maker_price = taker_price * (1 + self._min_profitability)
 
-            raw_size_limit = min(
-                taker_order_book_size_limit,
-                maker_balance_size_limit,
-                taker_balance_size_limit
+            if self._adjust_orders_enabled:
+                # If your ask is lower than the the top ask, increase it to just one tick below top ask
+                if maker_price < next_price_below_top_ask:
```

Comment:
Similarly, `maker_price = max(maker_price, next_price_below_top_ask)` would be better here.